src/Core/Configure/Engine/PhpConfig.php

```diff
@@ -93,6 +93,7 @@ public function read($key)
         if (!isset($config)) {
             throw new Exception(sprintf('Config file "%s" did not return an array', $key . '.php'));
         }
+        deprecationWarning('PHP configuration files should not set `$config. Instead return an array.');
 
```

Comment:
```php
if(!isset($config)){
   throw new Exception....
}else{
    throw new Warning......
}
```
what this say !?

Reply:
The code path for a file returning an array is on line 89 - just outside if this diff.